django/db/backends/base/schema.py

```diff
@@ -1601,14 +1601,29 @@ def _field_should_be_altered(self, old_field, new_field, ignore=None):
         # - changing an attribute that doesn't affect the schema
         # - changing an attribute in the provided set of ignored attributes
         # - adding only a db_column and the column name is not changed
+        # - field is related to a moving model and its db attrs are unchanged
         for attr in ignore.union(old_field.non_db_attrs):
             old_kwargs.pop(attr, None)
         for attr in ignore.union(new_field.non_db_attrs):
             new_kwargs.pop(attr, None)
+        if self._is_moving_model_rel_field(old_field, new_field):
+            old_kwargs.pop("to")
+            new_kwargs.pop("to")
         return self.quote_name(old_field.column) != self.quote_name(
             new_field.column
         ) or (old_path, old_args, old_kwargs) != (new_path, new_args, new_kwargs)
 
+    def _is_moving_model_rel_field(self, old_field, new_field):
+        if (old_field.remote_field and new_field.remote_field) and not (
+            isinstance(new_field.related_model, str)
+        ):
+            # Check if the field being altered is related to the moving model.
+            old_app_label = old_field.related_model._meta.app_label
+            new_app_label = new_field.related_model._meta.app_label
+            old_db_table = old_field.related_model._meta.db_table
+            new_db_table = new_field.related_model._meta.db_table
+            return old_app_label != new_app_label and old_db_table == new_db_table
```

Comment:
I wonder if we could do this as a separate cleanup and do we need to check app labels :thinking: What about a model with `db_table` that was renamed in the same app?
For example
```python
class MainModel(models.Model):
    class Meta:
        db_table = "custom_db_table"


class MyModel(models.Model):
    field_fk = models.ForeignKey('MainModel', on_delete=models.CASCADE)


class MyModelM2M(models.Model):
    field_m2m = models.ManyToManyField('MainModel')
```
was changed to the:
```python
class RenamedMainModel(models.Model):
    class Meta:
        db_table = "custom_db_table"


class MyModel(models.Model):
    field_fk = models.ForeignKey('RenamedMainModel', on_delete=models.CASCADE)


class MyModelM2M(models.Model):
    field_m2m = models.ManyToManyField('RenamedMainModel')
```
As far as I'm aware, this should be a no-op. Also `_is_moving_model_rel_field()` hooks seems unnecessary.

Reply:
> Do we also need a separate ticket for this?

We can do this as a cleanup related to the 24686: `Refs #24686 -- Made ...`

Reply:
okay 👍